emote.py

```diff
@@ -14,11 +14,15 @@ def init(self):
 		if self.path is not None:
 			try:
 				if not os.path.isfile(self.path):
-					open(self.path, 'w+')
+					with open(self.path, 'w+') as f:
+						json.dump(self.emotes, f)
 
 				else:
-					self.emotes = json.loads(open(self.path, 'r').read())
+					with open(self.path, 'r') as f:
+						self.emotes = json.loads(f.read())
 
+			except (JSONDecodeError):
```

Comment:
Just looked it up, apparently in python3.5, json.loads throws a JSONDecodeError if the object is not valid JSON. In python3.4, json.loads throws a ValueError.
https://docs.python.org/3/library/json.html